src/index.js

```diff
@@ -22,14 +22,19 @@ import history from "./lib/history";
 // Serviceworker import
 import * as serviceWorker from "./serviceWorker";
 
+// FeatureFlag imports
+import { FeatureFlagsProvider } from "./components/FeatureFlags";
+
 injectStore(store);
 
 ReactDOM.render(
   <React.StrictMode>
     <Provider store={store}>
       <PersistGate loading={null} persistor={persistor}>
         <HistoryRouter history={history}>
-          <App />
+          <FeatureFlagsProvider>
+            <App />
+          </FeatureFlagsProvider>
```

Comment:
I don't think writing your own context is a good idea. By default, React context is not persistent through page refreshes.

However, we do use Redux, and I have Redux state persistence infrastructure. A better idea would be storing/accessing the feature flags in Redux

Reply:
Also, BTW, Redux (and the redux state persistor we use) is built upon React Contexts. So essentially you are just switching to a different Context and getting many perks for free.

https://github.com/WilliamsStudentsOnline/wso-react/blob/4635ca3c39e90d2136f4a87ff132adee6b810c60/src/index.js#L13-L15
https://github.com/WilliamsStudentsOnline/wso-react/blob/4635ca3c39e90d2136f4a87ff132adee6b810c60/src/index.js#L32-L33